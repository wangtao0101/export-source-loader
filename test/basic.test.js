import compiler from './compiler.js';

describe('basic test', () => {
  test('basic loader', async () => {
    const stats = await compiler('./fixtures/basic.js');
    expect(stats.toJson().modules[0].source).toMatchSnapshot();
  });

  test('loader with babel-loader', async () => {
    const stats = await compiler('./fixtures/basic.js', { transformJS: true });
    expect(stats.toJson().modules[0].source).toMatchSnapshot();
  });

  test('loader with option module = commonjs', async () => {
    const stats = await compiler('./fixtures/basic.js', { loaderOptions: { module: 'commonjs' } });
    expect(stats.toJson().modules[0].source).toMatchSnapshot();
  });

  test('loader with option name', async () => {
    const stats = await compiler('./fixtures/basic.js', { loaderOptions: { name: 'sourceCode' } });
    expect(stats.toJson().modules[0].source).toMatchSnapshot();
  });

  test('loader with option module expect es and commonjs', async () => {
    const stats = await compiler('./fixtures/basic.js', { loaderOptions: { module: 'xxxx' } });
    expect(stats.toJson().modules[0].source).toMatchSnapshot();
  });

  test('loader with option name minLength < 5', async () => {
    const stats = await compiler('./fixtures/basic.js', { loaderOptions: { name: 'sou' } });
    expect(stats.toJson().modules[0].source).toMatchSnapshot();
  });
});
