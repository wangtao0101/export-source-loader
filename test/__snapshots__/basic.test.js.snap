// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic test basic loader 1`] = `
"const a = 1;
export default a;

export var source = \\"const%20a%20%3D%201%3B%0Aexport%20default%20a%3B%0A\\""
`;

exports[`basic test loader with babel-loader 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.source = exports.default = void 0;
var a = 1;
var _default = a;
exports.default = _default;
var source = \\"const%20a%20%3D%201%3B%0Aexport%20default%20a%3B%0A\\";
exports.source = source;"
`;

exports[`basic test loader with option module = commonjs 1`] = `
"const a = 1;
export default a;

module.exports.source = \\"const%20a%20%3D%201%3B%0Aexport%20default%20a%3B%0A\\""
`;

exports[`basic test loader with option module expect es and commonjs 1`] = `"throw new Error(\\"Module build failed: ValidationError: export source loader Invalid Options\\\\n\\\\noptions.module should be equal to one of the allowed values\\\\n\\");"`;

exports[`basic test loader with option name 1`] = `
"const a = 1;
export default a;

export var sourceCode = \\"const%20a%20%3D%201%3B%0Aexport%20default%20a%3B%0A\\""
`;

exports[`basic test loader with option name minLength < 5 1`] = `"throw new Error(\\"Module build failed: ValidationError: export source loader Invalid Options\\\\n\\\\noptions.name should NOT be shorter than 5 characters\\\\n\\");"`;

exports[`basic test test react component 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.source = exports.default = void 0;

var _react = _interopRequireDefault(require(\\"react\\"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _typeof(obj) { if (typeof Symbol === \\"function\\" && typeof Symbol.iterator === \\"symbol\\") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === \\"function\\" && obj.constructor === Symbol && obj !== Symbol.prototype ? \\"symbol\\" : typeof obj; }; } return _typeof(obj); }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError(\\"Cannot call a class as a function\\"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if (\\"value\\" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === \\"object\\" || typeof call === \\"function\\")) { return call; } if (self === void 0) { throw new ReferenceError(\\"this hasn't been initialised - super() hasn't been called\\"); } return self; }

function _inherits(subClass, superClass) { if (typeof superClass !== \\"function\\" && superClass !== null) { throw new TypeError(\\"Super expression must either be null or a function\\"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var Test =
/*#__PURE__*/
function (_React$Component) {
  _inherits(Test, _React$Component);

  function Test() {
    _classCallCheck(this, Test);

    return _possibleConstructorReturn(this, (Test.__proto__ || Object.getPrototypeOf(Test)).apply(this, arguments));
  }

  _createClass(Test, [{
    key: \\"render\\",
    value: function render() {
      return _react.default.createElement(\\"div\\", null, \\"asdf\\");
    }
  }]);

  return Test;
}(_react.default.Component);

exports.default = Test;
var source = \\"import%20React%20from%20%27react%27%3B%0A%0Aexport%20default%20class%20Test%20extends%20React.Component%20%7B%0A%20%20render%28%29%20%7B%0A%20%20%20%20return%20%28%0A%20%20%20%20%20%20%3Cdiv%3E%0A%20%20%20%20%20%20%20%20asdf%0A%20%20%20%20%20%20%3C/div%3E%0A%20%20%20%20%29%0A%20%20%7D%0A%7D%0A\\";
exports.source = source;"
`;
